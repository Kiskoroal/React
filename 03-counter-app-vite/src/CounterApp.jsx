import PropTypes from 'prop-types';
import { useState } from 'react';

export const CounterApp = ( {value} ) => {

    const [counter, setCounter] = useState(value);

    const click = () => {
      setCounter(counter + 1);  
    };

    const rest = () => {
        setCounter(counter - 1);
    }

    const reset = () => {
        setCounter(value);
    }

    return(
        <div data-testid = 'test-counter'>
        <h1>Cookie Clicker</h1>
        <h2> {counter} </h2>

        <button onClick={ click }> +1 </button>
        <button onClick={ rest }> -1 </button>
        <button onClick={ reset }> Reset </button>
        </div>
    );
}

CounterApp.propTypes = {
    counter : PropTypes.number
};