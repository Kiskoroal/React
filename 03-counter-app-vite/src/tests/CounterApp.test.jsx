/** @jest-environment jsdom */
import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from '../CounterApp';

describe('Prueba de componente CounterApp', () => {

    const initialValue = 10;

    test('hacer match con snap', () => {
        const { container } = render(<CounterApp value={initialValue} />)

        expect(container).toMatchSnapshot();
    })
    

    test('debe incrementar con el boton +1', () => {
        
        render( <CounterApp value ={initialValue}  /> );
        fireEvent.click( screen.getByText('+1') );
        expect( screen.getByText('11') ).toBeTruthy();
    })
    
    test('debe decrementar con el boton -1', () => {
        
        render(<CounterApp value={initialValue} />);
        fireEvent.click(screen.getByText('-1'));
        expect( screen.getByText('9') ).toBeTruthy();
    })
    
    test('debe de funcionar el boton reset', () => {
        render(<CounterApp value={355} />);
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('Reset') );

        expect( screen.getByText(355) ).toBeTruthy();
    });

});