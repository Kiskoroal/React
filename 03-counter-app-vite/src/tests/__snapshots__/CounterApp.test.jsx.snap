// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de componente CounterApp hacer match con snap 1`] = `
<div>
  <div
    data-testid="test-counter"
  >
    <h1>
      Cookie Clicker
    </h1>
    <h2>
       
      10
       
    </h2>
    <button>
       +1 
    </button>
    <button>
       -1 
    </button>
    <button>
       Reset 
    </button>
  </div>
</div>
`;
